cms/models.py

```diff
@@ -253,6 +258,73 @@ class FacultyMembersPage(CourseProgramChildPage):
     ]
 
 
+class HomePage(Page):
+    """
+    CMS Page representing the home/root route
+    """
+
+    # disable promote panels, no need for slug entry, it will be autogenerated
+    promote_panels = []
+
+    subpage_types = [
+        "CoursePage",
+        "ProgramPage",
+        "CoursesInProgramPage",
+        "LearningTechniquesPage",
+        "UserTestimonialsPage",
+        "WhoShouldEnrollPage",
+        "FacultyMembersPage",
+    ]
+
+    def _get_child_page_of_type(self, cls):
+        """Gets the first child page of the given type if it exists"""
+        child = self.get_children().type(cls).first()
+        if child:
+            return child.specific
+        return None
+
+    @property
+    def learning_experience(self):
+        """
+        Gets the "Learning Experience" section subpage
+        """
+        return self._get_child_page_of_type(LearningTechniquesPage)
+
+    @property
+    def testimonials(self):
+        """
+        Gets the testimonials section subpage
+        """
+        return self._get_child_page_of_type(UserTestimonialsPage)
+
+    @property
+    def upcoming_courseware(self):
+        """
+        Gets the upcoming courseware section subpage
+        """
+        return self._get_child_page_of_type(CoursesInProgramPage)
+
+    def save(self, *args, **kwargs):
+        # autogenerate a unique slug so we don't hit a ValidationError
+        self.slug = slugify("home-{}".format(self.title))
+        super().save(*args, **kwargs)
+
+    def get_context(self, request, *args, **kwargs):
+        from courses.models import Program
+
+        context = super().get_context(request)
+        context.update(**get_js_settings_context(request))
+
+        # This just a test to populate the "Upcoming Courses and Programs"
+        # carousel. Actual implementation for this will come later
+        context.update(
+            courses=Program.objects.filter(title="SEED Digital Learning")
+            .first()
+            .courses.all()
+        )
```

Comment:
When we will add actual implementation of context ? In this PR ?

Reply:
Yes. In case you need to work on some issue based on this one, you can branch off of this PR and merge once this goes through.

Reply:
So this will be fixed as part of #252?

Reply:
Yes